### PR TITLE
Fix/multi project leaking

### LINKED ISSUE #162: Issues using multi mod setup with eclipse
Hey there,
I'm having some difficulties setting up my multi-mod workspace in 1.20.6. For some reason the neoforge jar dependency is added twice for each project causing a couple of issues inside eclipse but also when running the game giving the following error:

```
Exception in thread "main" net.neoforged.fml.ModLoadingException: Loading errors encountered: [
	fml.modloading.multiple_copies_on_classpath
]
	at MC-BOOTSTRAP/fml_loader@3.0.19/net.neoforged.fml.util.DevEnvUtils.findFileSystemRootOfFileOnClasspath(DevEnvUtils.java:78)
	at MC-BOOTSTRAP/fml_loader@3.0.19/net.neoforged.fml.loading.targets.CommonUserdevLaunchHandler.collectAdditionalModFileLocators(CommonUserdevLaunchHandler.java:24)
	at MC-BOOTSTRAP/fml_loader@3.0.19/net.neoforged.fml.loading.FMLLoader.beginModScan(FMLLoader.java:131)
	at MC-BOOTSTRAP/fml_loader@3.0.19/net.neoforged.fml.loading.FMLServiceProvider.beginScanning(FMLServiceProvider.java:76)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.TransformationServiceDecorator.runScan(TransformationServiceDecorator.java:95)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.TransformationServicesHandler.lambda$runScanningTransformationServices$7(TransformationServicesHandler.java:94)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.HashMap$ValueSpliterator.forEachRemaining(HashMap.java:1787)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.TransformationServicesHandler.runScanningTransformationServices(TransformationServicesHandler.java:96)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.TransformationServicesHandler.initializeTransformationServices(TransformationServicesHandler.java:50)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.Launcher.run(Launcher.java:84)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.Launcher.main(Launcher.java:74)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.2/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23)
	at cpw.mods.bootstraplauncher@2.0.0/cpw.mods.bootstraplauncher.BootstrapLauncher.run(BootstrapLauncher.java:176)
	at cpw.mods.bootstraplauncher@2.0.0/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:65)
```

When viewing the projects 'Build Path', this dependency can be found twice:
image

My setup can be found here: https://github.com/CreativeMD/ForgeMods/tree/1.20.5

### PR DESCRIPTION
**Pull request by marchermans**
### TLDR
Due to the way we register dependencies on replacement it is possible that the target configuration still leaks information.
This replaces this by properly looking up the relevant `compileOnly` and `runtimeOnly` configurations.

### Changes:
- Fixes to how configurations are looked up from a root configuration (aka implementation -> compileOnly etc)
- Implement test that confirms the replacement behaviour.

### Issues:
Closes: #162

### CODE REVIEW

File: dsl/common/src/main/groovy/net/neoforged/gradle/dsl/common/util/GradleInternalUtils.groovy [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * Gets the name of the source set with the given post fix
+     *
+     * @param sourceSet The source set to get the name of
+     * @param postFix The post fix to append to the source set name
+     * @return The name of the source set with the post fix
+     */
+    static String getSourceSetName(SourceSet sourceSet, String postFix) {
+        if (sourceSet instanceof DefaultSourceSet) {
```
Comment (Technici4n) [inline dsl/common/src/main/groovy/net/neoforged/gradle/dsl/common/util/GradleInternalUtils.groovy]: I'm not a fan of this. It's annoying that they don't expose this, but in principle the logic below should work for all source sets anyway?

**Review by Technici4n — APPROVED** [review]
